llama_stack/apis/post_training/post_training.py

```diff
@@ -184,7 +182,7 @@ async def supervised_fine_tune(
             description="Model descriptor from `llama model list`",
         ),
         checkpoint_dir: Optional[str] = None,
-        algorithm_config: Optional[LoraFinetuningConfig | QATFinetuningConfig] = None,
+        algorithm_config: Optional[_AlgorithmConfig] = None,
```

Comment:
ah I see why you made the changes. I think this signature was wrong previously. We should just use `AlgorithmConfig` here and avoid having to do any `_AlgorithmConfig` alias etc.

Reply:
I think mypy is confused by `register_schema` generics; if I directly use the result of the function call as a type, I get the following error:

```
llama_stack/providers/inline/post_training/torchtune/post_training.py:57: error: Variable "llama_stack.apis.post_training.post_training.AlgorithmConfig" is not valid as a type[valid-type]
llama_stack/providers/inline/post_training/torchtune/post_training.py:57: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
```

Reply:
@booxter do TypeAliases work? 
```python
from typing import TypeAlias

AlgorithmConfig: TypeAlias = register_schema(
  ... 
)
```

Reply:
I don't think we actually need any of that. The trick is to *not* set the variable to the return value of the function. We can just define the type; then call `register_schema`, ignoring its return value (since it's the exact same value passed as `data_type` argument). I updated the patch, let me know what you think.

Reply:
If this makes sense, we can convert everything in the tree to the same approach: https://github.com/meta-llama/llama-stack/pull/1713